cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -888,14 +895,228 @@ public void testBuildBalProjectWithCustomMavenRepo() throws IOException {
+        Path projectPath = testResources.resolve("dep-dist-version-projects").resolve("preCompiledPackage");
+        replaceDependenciesTomlContent(projectPath, "**INSERT_DISTRIBUTION_VERSION_HERE**", "2201.5.0");
+        System.setProperty("user.dir", projectPath.toString());
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+
+        new CommandLine(buildCommand).parseArgs();
+        buildCommand.execute();
+        String buildLog = readOutput(true);
+        Assert.assertEquals(
+                buildLog.replaceAll("\r", ""),
+                getOutput("build-old-dist-precomp-proj-with-dep.txt")
+                        .replaceAll("INSERT_NEW_DIST_VERSION_HERE", getNewVersionForOldDistWarning())
+                        .replaceAll("INSERT_OLD_DIST_VERSION_HERE", getOldVersionForOldDistWarning("2201.5.0")));
+
+        // Dependencies.toml should be updated to the latest minor versions.
+        // depA:1.0.0 -> depA:1.1.0
+        // depB:1.0.0 -> depB:1.1.0
+        Path actualDependenciesToml = projectPath.resolve("Dependencies.toml");
+        Path expectedDependenciesToml = testResources.resolve("dep-dist-version-projects").resolve("expected-dep-tomls")
+                .resolve("old-dist-precomp-pkg-with-no-sticky.toml");
+        Assert.assertTrue(actualDependenciesToml.toFile().exists());
+        assertTomlFilesEquals(actualDependenciesToml, expectedDependenciesToml);
+
+        // Revert the distribution version and dependency versions to the placeholder values
+        replaceDependenciesTomlContent(projectPath, RepoUtils.getBallerinaShortVersion(),
+                "**INSERT_DISTRIBUTION_VERSION_HERE**");
+        replaceDependenciesTomlContent(projectPath, "1.1.0", "1.0.0");
+        deleteDirectory(projectPath.resolve("target"));
+    }
+
+    @Test(description = "Build a project already built with an older distribution with sticky flag",
+            groups = {"proj-with-deps-update-policy"})
+    public void testBuildProjectPrecompiledWithOlderDistWithStickyFlag() throws IOException {
+        Path projectPath = testResources.resolve("dep-dist-version-projects").resolve("preCompiledPackage");
+        replaceDependenciesTomlContent(projectPath, "**INSERT_DISTRIBUTION_VERSION_HERE**", "2201.5.0");
+        System.setProperty("user.dir", projectPath.toString());
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+
+        new CommandLine(buildCommand).parseArgs("--sticky");
+        buildCommand.execute();
+        String buildLog = readOutput(true);
+        Assert.assertEquals(
+                buildLog.replaceAll("\r", ""),
+                getOutput("build-old-dist-precomp-proj-with-dep.txt")
+                        .replaceAll("INSERT_NEW_DIST_VERSION_HERE", getNewVersionForOldDistWarning())
+                        .replaceAll("INSERT_OLD_DIST_VERSION_HERE", getOldVersionForOldDistWarning("2201.5.0")));
```

Comment:
The output seems to be wrong. For the combination where Dependencies.toml exists and --sticky flag is given, we don't update automatically. Instead, we give a hint asking the user to set sticky to false.

Reply:
Here the Dependencies.toml sticks to the dependencies correctly. But the warning is inaccurate. will update the logic to have a hint as you suggested instead of the warning.

Reply:
The hint was produced correctly when there no distribution-version field in the Dependencies toml. It was not produced when there is a distribution-version field but it is an older version than what the user currently has. This is fixed now.